<launch>

<!-- ROS建图或导航都只支持一个雷达，这个时候就需要做2个雷达的融合。
思路：先将两个激光雷达获得的laser_scan转成point_cloud点云，利用pcl库将两个点云拼接在一起，然后在把拼接后的点云重新转成laser_scan
关键点：两个激光雷达的偏移量、雷达时间同步 -->

  <arg name="if_double" default="true"/>
  <!-- <arg name="lidar1_rotate"  default="-135"/>
  <arg name="lidar1_xoffset" default="0.3"/>
  <arg name="lidar1_yoffset" default="0.235"/>
  <arg name="lidar2_rotate"  default="45"/>
  <arg name="lidar2_xoffset" default="-0.3"/>
  <arg name="lidar2_yoffset" default="-0.235"/> -->
  <!-- front_lidar -->
  <!-- 我们用的是仿真雷达，坐标系固定了，无需旋转坐标系 -->
  <arg name="lidar1_rotate"  default="0"/>
  <!-- 距离base_link（原点）的偏移量 -->
  <arg name="lidar1_xoffset" default="0.3"/>
  <arg name="lidar1_yoffset" default="0.235"/>
  <!-- back_lidar -->
  <arg name="lidar2_rotate"  default="0"/>
  <arg name="lidar2_xoffset" default="-0.3"/>
  <arg name="lidar2_yoffset" default="-0.235"/>

  <!-- <node if="$(arg if_double)" pkg="lslidar_x10_driver" type="lslidar_x10_driver_node" name="M10_P_1" output="screen">
    <param name="lidar_name" value="M10_P"/>                        #雷达选择:M10 M10_P M10_PLUS M10_GPS N10
    <param name="serial_port" value="/dev/wheeltec_lidar1"/>        #雷达连接的串口
    <param name="interface_selection" value="serial"/>              #接口选择:net 为网口,serial 为串口。
    <param name="frame_id" value="laser1"/>                         #激光坐标
    <param name="min_distance" type="double" value="0"/>            #雷达接收距离最小值
    <param name="max_distance" type="double" value="100"/>          #雷达接收距离最大值
    <param name="scan_topic" value="scan1"/>                        #设置激光数据topic名称
    <param name="use_gps_ts" value="false"/>                        #雷达是否使用GPS授时
  </node> -->

  <!-- <node if="$(arg if_double)" pkg="lslidar_x10_driver" type="lslidar_x10_driver_node" name="M10_P_2" output="screen">
    <param name="lidar_name" value="M10_P"/>                        #雷达选择:M10 M10_P M10_PLUS M10_GPS N10
    <param name="serial_port" value="/dev/wheeltec_lidar2"/>        #雷达连接的串口
    <param name="interface_selection" value="serial"/>              #接口选择:net 为网口,serial 为串口。
    <param name="frame_id" value="laser2"/>                         #激光坐标
    <param name="min_distance" type="double" value="0"/>            #雷达接收距离最小值
    <param name="max_distance" type="double" value="100"/>          #雷达接收距离最大值
    <param name="scan_topic" value="scan2"/>                        #设置激光数据topic名称
    <param name="use_gps_ts" value="false"/>                        #雷达是否使用GPS授时
  </node> -->

  <node if="$(arg if_double)" name="laser_double" pkg="chibot_slam" type="laser_double" output="screen">
    <param name="frame_id"        type="string"   value="laser"/> <!--雷达输出frame_id-->
    <param name="scan_topic"      type="string"   value="scan"/> <!--雷达输出话题-->
    <!--0:两雷达不屏蔽角度 1:两雷达屏蔽角度-->
    <!-- <param name="truncated_mode" value="1"/> -->
    <!-- 我们不屏蔽角度，因为我们在仿真雷达插件中设置了有限扫描角度 -->
    <param name="truncated_mode" value="0"/>

    <param name="scan1_topic"      type="string"   value="scan1"/>   <!-- 雷达1话题-->
    <param name="lidar1_rotate"    type="double"   value="$(arg lidar1_rotate)"/> <!--雷达1坐标旋转角度,小车前方为0度角，正方向为顺时针-->
    <param name="lidar1_xoffset"   type="double"   value="$(arg lidar1_xoffset)"/> <!-- 雷达1坐标原点x轴偏移距离-->
    <param name="lidar1_yoffset"   type="double"   value="$(arg lidar1_yoffset)"/> <!-- 雷达1坐标原点y轴偏移距离-->
    <rosparam param="disable_min1">[120]</rosparam>                      <!--雷达1屏蔽角度左值（角度顺时针方向，支持多角度屏蔽）-->
    <rosparam param="disable_max1">[240]</rosparam>                     <!--雷达1屏蔽角度右值（角度顺时针方向，支持多角度屏蔽）-->

    <param name="scan2_topic"      type="string"   value="scan2"/>  <!-- 雷达2话题-->
    <param name="lidar2_rotate"    type="double"   value="$(arg lidar2_rotate)"/> <!--雷达2坐标旋转角度,小车前方为0度角，正方向为顺时针-->
    <param name="lidar2_xoffset"   type="double"   value="$(arg lidar2_xoffset)"/>      <!-- 雷达2坐标原点x轴偏移距离-->
    <param name="lidar2_yoffset"   type="double"   value="$(arg lidar2_yoffset)"/>     <!-- 雷达2坐标原点y轴偏移距离-->
    <rosparam param="disable_min2">[120]</rosparam> <!--雷达2屏蔽角度左值（角度顺时针方向，支持多角度屏蔽）-->
    <rosparam param="disable_max2">[240]</rosparam> <!--雷达2屏蔽角度右值（角度顺时针方向，支持多角度屏蔽）-->
  </node>

</launch>
